### PR TITLE
IOS-14409: expose sharedPaginatedRequestData of BOXFolderItemsRequest to link related paginated requests

### PR DESCRIPTION
**Pull request by thuyBox**
and add cache client protocol for completion of BOXFolderItemsRequest

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @thuyBox has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h [hunk 1]
```diff
@@ -213,6 +213,9 @@
                               withFolder:(BOXFolder *)folder
                                    error:(NSError *)error;
 
+- (void)deleteChildrenOfFolderWithID:(NSString *)folderID
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h]: Should it be associated with a request? This feels odd of place, can we see if it should conform more to the existing pattern or whether it belongs somewhere else?

Comment (thuyBox) [reply BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h]: done

**Review by crousselle — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h [hunk 2]
```diff
@@ -213,6 +213,8 @@
                               withFolder:(BOXFolder *)folder
                                    error:(NSError *)error;
 
+- (void)finishFolderItemsRequest:(BOXFolderItemsRequest *)request;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Protocols/BOXContentCacheClientProtocol.h]: Minor: naming, hasFinishedFolderItemsRequest? Just to make it clear that it isn't a call to action, but rather information that an event has occurred.

**Review by boxtcarpel — APPROVED** [review]

**Pull request merged**